torchaudio/functional/functional.py

```diff
@@ -1377,9 +1378,16 @@ def resample(
             but less efficient. We suggest around 4 to 10 for normal use. (Default: ``6``)
         rolloff (float, optional): The roll-off frequency of the filter, as a fraction of the Nyquist.
             Lower values reduce anti-aliasing, but also reduce some of the highest frequencies. (Default: ``0.99``)
+        kernel (Tensor, optional): Tensor of dimension (f, 1, w) representing the windowed sinc function that is
```

Comment:
1. I don't think passing in width will make computation flow (much) simpler -- my concern is that different libraries may compute kernels differently or use different convolution sizes (which is what width is being used for), and it is therefore hard to do sanity checks on their input to make sure it is reasonable and aligns with their parameters. Also because there is simplification to be done to create these kernels (like dividing by gcd) and it isn't super intuitive, I don't think kernel should be a parameter that should be exposed to users (unless we also expose the kernel function and provide concrete documentation for it)
2. `%timeit torchaudio.functional.resample(sig_torch, P, Q)` gives `10.6 ms ± 218 µs per loop` 
`resampler = torchaudio.transforms.Resample(); %timeit resampler(sig_torch)` gives `1.46 ms ± 95.6 µs per loop`

Reply:
We may want to introduce a new Transform for fixed sampling rate, for faster computation.

Reply:
Yes, the kernel is dependent on a set of parameters (old_freq, new_freq, lowpass_filter_width, rolloff,..) and is the same for that given set of parameters. Currently in transforms.Resample, all of these parameters are being passed in init, and the only parameter for forward is the waveform itself, so I think it is fine to reuse the current transform.